R/mwrite.R

```diff
@@ -0,0 +1,341 @@
+.quotes <- function(x) paste0("\"", x, "\"")
+
+tocode <- function(l) {
+  if(!length(l)) return(character(0))
+  sc <- sapply(l, function(x) is.character(x) && length(x)==1)
+  l[sc] <- sapply(l[sc], .quotes)
+  paste0(names(l), " = ", as.character(l))
+}
+
+get_upper_tri <- function(x) {
+  x <- as.matrix(x)
+  x[upper.tri(x, diag = TRUE)]
+}
+
+require_yaml <- function() {
+  if(!requireNamespace("yaml", quietly = TRUE)) {
+    abort("The package \"yaml\" is required.") # nocov
+  } 
+}
+
+mwrite_model_to_list <- function(x) {
+  l <- list()
+  # Header
+  l$source <- "mrgsolve::mwrite"
+  l$mrgsolve <- as.character(packageVersion("mrgsolve"))
+  l$format <- "list"
+  l$version <- 1
+  l$model <- x@model
+  # Problem
+  l$prob <- ""
+  # Critical items
+  l$param <- as.list(param(x))
+  l$init <- as.list(init(x))
+  l$capture <- character(0)
+  if(length(x@capture)) {
+    capture <- names(x@capture)
```

Comment:
Some funny handling here ... 

$CAPTURE items can be renamed as

```
$CAPTURE new = old
```

what we store in `x@capture` is the new name, with the old names in the `names` attribute.

Reply:
It looks like this

```r
> mcode("test", "$PARAM old = 2\n$CAPTURE new = old", compile = FALSE)@capture
  old 
"new" 
```